### PR TITLE
Remove CONFIG_CLOCK_MONOTONIC special code

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since this option doesn't exist anymore.
Depends on https://github.com/apache/incubator-nuttx/pull/5503, both patch need merge together.

## Impact
Minor

## Testing
Pass CI

### CODE REVIEW

**Comment by acassis** [comment]
@xiaoxiang781216 what happened to CLOCK_REALTIME? If I'm not wrong CLOCK_REALTIME is affected when you change the time/date and CLOCK_MONOTONIC is not. Isn't going break some application and/or Linux/POSIX compatibility? For measuring elapsed time in fact CLOCK_MONOTONIC is the solution, but I guess there are cases where CLOCK_REALTIME could be useful. More info: https://stackoverflow.com/questions/3523442/difference-between-clock-realtime-and-clock-monotonic http://www.qnx.com/developers/docs/qnxcar2/index.jsp?topic=%2Fcom.qnx.doc.neutrino.getting_started%2Ftopic%2Fs1_timer_CLOCK_MONOTONIC.html https://tagmerge.com/question/difference-between-clock_realtime-and-clock_monotonic?answer=61e1f9684f0d313df01d4719

**Comment by hartmannathan** [comment]
> @xiaoxiang781216 what happened to CLOCK_REALTIME? If I'm not wrong CLOCK_REALTIME is affected when you change the time/date and CLOCK_MONOTONIC is not. Isn't going break some application and/or Linux/POSIX compatibility? For measuring elapsed time in fact CLOCK_MONOTONIC is the solution, but I guess there are cases where CLOCK_REALTIME could be useful. More info: https://stackoverflow.com/questions/3523442/difference-between-clock-realtime-and-clock-monotonic http://www.qnx.com/developers/docs/qnxcar2/index.jsp?topic=%2Fcom.qnx.doc.neutrino.getting_started%2Ftopic%2Fs1_timer_CLOCK_MONOTONIC.html https://tagmerge.com/question/difference-between-clock_realtime-and-clock_monotonic?answer=61e1f9684f0d313df01d4719

CLOCK_REALTIME can go "backwards" due to clock changes, daylight saving time, leap seconds, NTP, and other reasons. For measuring elapsed time you definitely want CLOCK_MONOTONIC.

From memory, I think both clocks are present in POSIX at minimum; then there are other clocks like, e.g., CLOCK_BOOTTIME that are not guaranteed to be present.

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 what happened to CLOCK_REALTIME? If I'm not wrong CLOCK_REALTIME is affected when you change the time/date and CLOCK_MONOTONIC is not. Isn't going break some application and/or Linux/POSIX compatibility? For measuring elapsed time in fact CLOCK_MONOTONIC is the solution, but I guess there are cases where CLOCK_REALTIME could be useful. More info: https://stackoverflow.com/questions/3523442/difference-between-clock-realtime-and-clock-monotonic http://www.qnx.com/developers/docs/qnxcar2/index.jsp?topic=%2Fcom.qnx.doc.neutrino.getting_started%2Ftopic%2Fs1_timer_CLOCK_MONOTONIC.html https://tagmerge.com/question/difference-between-clock_realtime-and-clock_monotonic?answer=61e1f9684f0d313df01d4719

reply in https://github.com/apache/incubator-nuttx/pull/5503

**Comment by xiaoxiang781216** [comment]
ping @acassis

**Review by pkarashchenko — APPROVED** [review]

File: examples/calib_udelay/Kconfig [hunk 1]
```diff
@@ -6,7 +6,6 @@
 config EXAMPLES_CALIB_UDELAY
 	tristate "Calibration tool for udelay"
 	default n
-	select CLOCK_MONOTONIC
```
Comment (normanr) [inline examples/calib_udelay/Kconfig:9]: There was no change to remove CLOCK_MONOTONIC from calib_udelay_main.c, so this breaks the build.

File: examples/canard/Kconfig [hunk 2]
```diff
@@ -6,7 +6,7 @@
 config EXAMPLES_LIBCANARD
 	tristate "libcandard example"
 	default n
-	depends on CANUTILS_LIBCANARD && CLOCK_MONOTONIC && SYSTEM_TIME64
+	depends on CANUTILS_LIBCANARD && SYSTEM_TIME64
```
Comment (normanr) [inline examples/canard/Kconfig:9]: There was no change to remove CLOCK_MONOTONIC from canard_main.c, so this breaks the build.

File: examples/calib_udelay/Kconfig (hunk 1 above)
Comment (xiaoxiang781216) [reply examples/calib_udelay/Kconfig:9]: CLOCK_MONOTONIC always exist after https://github.com/apache/incubator-nuttx/pull/5503, so CLOCK_MONOTONIC can be used without any precondition.

File: examples/canard/Kconfig (hunk 2 above)
Comment (xiaoxiang781216) [reply examples/canard/Kconfig:9]: ditto

File: examples/calib_udelay/Kconfig (hunk 1 above)
Comment (normanr) [reply examples/calib_udelay/Kconfig:9]: Thanks for the quick response. My nuttx and apps git trees were out-of-sync and syncing them to the same place fixed this.

**Pull request merged**